hudi-client/hudi-client-common/src/test/java/org/apache/hudi/testutils/Assertions.java

```diff
@@ -51,4 +67,88 @@ public static void assertFileSizesEqual(List<WriteStatus> statuses, CheckedFunct
         assertEquals(fileSizeGetter.apply(status), status.getStat().getFileSizeInBytes())));
   }
 
+  public static void assertPartitionMetadataForRecords(String basePath, List<HoodieRecord> inputRecords,
+                                                HoodieStorage storage) throws IOException {
+    Set<String> partitionPathSet = inputRecords.stream()
+            .map(HoodieRecord::getPartitionPath)
+            .collect(Collectors.toSet());
+    assertPartitionMetadata(basePath, partitionPathSet.stream().toArray(String[]::new), storage);
```

Comment:
For this line and 83, you can simplify this by not collecting to a set and just use `distinct()` on the stream

Reply:
done